### PR TITLE
[Docs] Add section on config validation to our CI/CD reference section

### PR DESCRIPTION
**Pull request by sgibson91**

### CODE REVIEW

File: docs/reference/ci-cd.md [hunk 1]
```diff
@@ -31,3 +37,24 @@ clusters.
...
+Here is an example of a [`values.schema.yaml` file from the basehub chart](https://github.com/2i2c-org/infrastructure/blob/HEAD/helm-charts/basehub/values.schema.yaml).
+These schemas explicitly list what values can be passed through our config, and what type these values should have.
+Therefore, we can use these schemas to validate all our config before making deploys, and catch bugs early.
+
+We have [functions within the deployer](https://github.com/2i2c-org/infrastructure/blob/HEAD/deployer/__main__.py#L213-L302) that validate the cluster config, the support chart values, and the helm chart values for each hub against these schemas.
+We automatically run these functions in GitHub Actions configured by the [`validate-clusters.yaml` workflow file](https://github.com/2i2c-org/infrastructure/blob/master/.github/workflows/validate-clusters.yaml).
+This workflow is triggered when changes to files along the following filepaths are detected.
```
Comment (consideRatio) [inline docs/reference/ci-cd.md:50]: I think listing these files isn't sustainable. It breaks the DRY principle without good enough motivation i think.

Whenever we update one we should update the other now or get outdated docs etc. I think if there should be an explicit comment on the level on when the workflow is triggered, it should be generally descriptive to say "only when related configuration has changed" or similar, omitting the exact details.

File: docs/reference/ci-cd.md [hunk 2]
```diff
@@ -31,3 +37,24 @@ clusters.
...
+
+```
+- config/clusters/**
+- deployer/**
+- helm-charts/basehub/**
+- helm-charts/daskhub/**
+- helm-charts/support/**
+- requirements.txt
+- .github/workflows/validate-hubs.yaml
+```
```
Comment (sgibson91) [inline docs/reference/ci-cd.md:60]:
```suggestion
This workflow is only triggered when related configuration has changed.
```

**Review by consideRatio — APPROVED** [review]
🎉🎉🎉🎉🎉🎉🎉

**Review by GeorgianaElena — APPROVED** [review]
:rocket:

**Pull request merged**